libtcmu.c

```diff
@@ -277,78 +268,43 @@ static int generic_handle_cmd(struct tcmu_device *dev,
 	case MODE_SELECT:
 	case MODE_SELECT_10:
 		return tcmu_emulate_mode_select(cdb, iovec, iov_cnt, sense);
+	default:
+		tcmu_err("unknown command %x\n", cdb[0]);
+		return TCMU_NOT_HANDLED;
+	}
+}
+
+static bool command_is_store_call(uint8_t cmd)
+{
```

Comment:
Maybe we are talking about different things, because I am confused about what the passthrough code has to do with this patchset.

These patches are laying the ground work for emulated commands that require multiple read/write calls but some handlers implement sync read/write callouts and some implement async ones. For example, glfs's read/write callouts are sync, and in Venky's PR 89 he has modified rbd's read/write callouts to use librbd's aio calls.

To support these two types of implementations for a command like WRITE_VERIFY, what is your proposal if not this patchset. For this command we want to write N bytes, then read N bytes and then compare the data to make sure it got written correctly, so libtcmu's tcmu_emulate_write_verify call needs to do a write into the handler then a read.

We could do:

1. Venky's patches here where libtcmu allows the generic write verify code to call the read or write callouts, and not worry about if they are sync or async. The store code figures it out. It assumes async and makes sync handler's calls look async.

2. The opposite of #1 and instead of making the handlers that have sync read/write callout implemenations look like async calls, we add wrappers around the handlers that have async read/write callout implementations that make them look like sync calls.

3. Not sure if this where the confusion is, but are you are saying each handler's passthrough code would just handle these types of commands? So glfs's handle_cmd would do a glfs_write/ and read for WRITE_VERIFY. rbd would do rbd_write and rbd_read.